src/libraries/Microsoft.Extensions.Hosting/src/Internal/HostingEnvironment.cs

```diff
@@ -15,7 +15,7 @@ public class HostingEnvironment : IHostingEnvironment, IHostEnvironment
     {
         public string EnvironmentName { get; set; } = null!;
 
-        public string? ApplicationName { get; set; }
+        public string ApplicationName { get; set; } = string.Empty;
```

Comment:
Nit: Should we also update `EnvironmentName` to initially be set to "Production" for consistency in how we implement the non-nullable default value?

Or update the assignment of this to be `ApplicationName = Assembly.GetEntryAssembly()?.GetName().Name ?? string.Empty` for consistency the other way?

Reply:
I don't really like having "policy" here... so instead to be consistent I changed all the non-nullable `string`s in this class to default to `string.Empty`.